plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStore.java

```diff
@@ -70,6 +69,11 @@ public S3BlobStore(Settings settings, AmazonS3 client, String bucket, @Nullable
             throw new BlobStoreException("Detected a buffer_size for the S3 storage lower than [" + MIN_BUFFER_SIZE + "]");
         }
 
+        this.cannedACL = initCannedACL(cannedACL);
+        if (this.cannedACL == null) {
+            throw new BlobStoreException("cannedACL is not valid: [" + cannedACL + "]");
```

Comment:
I'd throw the exception in `initCannedACL()` method instead of checking for a `null` here.